sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -228,4 +259,179 @@ public static TextAnalyticsException toTextAnalyticsException(
             .setCountryHint(textDocumentInput.getCountryHint())));
         return multiLanguageInputs;
     }
+
+    /**
+     * Extracts the result ID from the URL.
+     *
+     * @param operationLocation The URL specified in the 'Operation-Location' response header containing the
+     * resultId used to track the progress and obtain the result of the analyze operation.
+     *
+     * @return The resultId used to track the progress.
+     */
+    public static String parseModelId(String operationLocation) {
+        if (!CoreUtils.isNullOrEmpty(operationLocation)) {
+            int lastIndex = operationLocation.lastIndexOf('/');
+            if (lastIndex != -1) {
+                return operationLocation.substring(lastIndex + 1);
+            }
+        }
+        throw LOGGER.logExceptionAsError(
+            new RuntimeException("Failed to parse operation header for result Id from: " + operationLocation));
+    }
+
+    /**
+     * Extract the next pagination link which contains the request parameter values, into map,
+     * such as '$skip=20' and '$top=2'.
```

Comment:
I know you are passing showStats around, but this link will also contain it, so you could always aprse it here and use it

Reply:
parse the nextLink to get the stats=true/false value? Is it use the user's input stats parameter is easier?